tools/minimal_runtime_shell.py

```diff
@@ -125,20 +143,26 @@ def generate_minimal_runtime_load_statement(target_basename):
     else:
       return script_xhr + files_to_load[0] + ";"
 
-  if not shared.Settings.MODULARIZE:
+  if not shared.Settings.MODULARIZE or shared.Settings.USE_WASM_WORKERS:
     # If downloading multiple files like .wasm or .mem, those need to be loaded in
     # before we can add the main runtime script to the DOM, so convert the main .js
     # script load from direct script() load to a binary() load so we can still
     # immediately start the download, but can control when we add the script to the
     # DOM.
-    if shared.Settings.USE_PTHREADS:
+    if shared.Settings.USE_PTHREADS or shared.Settings.USE_WASM_WORKERS:
       script_load = "script(url)"
     else:
       script_load = "script(url).then(() => { URL.revokeObjectURL(url) });"
 
+    if shared.Settings.USE_WASM_WORKERS:
+      save_js = '%s.js = ' % shared.Settings.EXPORT_NAME
+    else:
+      save_js = ''
+
     files_to_load[0] = "binary('%s')" % (target_basename + '.js')
-    then_statements += ["var url = URL.createObjectURL(new Blob([r[0]], { type: 'application/javascript' }));",
-                        script_load]
+    if not shared.Settings.MODULARIZE:
+      then_statements += ["var url = %sURL.createObjectURL(new Blob([r[0]], { type: 'application/javascript' }));" % save_js,
+                          script_load]
```

Comment:
* We should adjust the download sequence so that the necessary .js files are kept in memory and loaded up by the Workers without pinging the remote server for each Worker startup. (that is dramatically slower e.g. for 16+ workers)